include/status-code/status_code.hpp

```diff
@@ -471,7 +524,7 @@ template <class DomainType> class SYSTEM_ERROR2_TRIVIAL_ABI status_code : public
   Does not check if domains are equal.
   */
   SYSTEM_ERROR2_TEMPLATE(class ErasedType)  //
-  SYSTEM_ERROR2_TREQUIRES(SYSTEM_ERROR2_TPRED(detail::type_erasure_is_safe<ErasedType, value_type>::value))
+  SYSTEM_ERROR2_TREQUIRES(SYSTEM_ERROR2_TPRED(traits::is_type_erasable_to<status_code<erased<ErasedType>>, value_type>::value))
```

Comment:
I would feel caution about use of final types in traits if it can be avoided i.e. `status_code`. They can cause a multipass evaluation in newer compilers, and in older compilers they are buggy in various unpleasant ways. This is why the original code didn't use final types, I think much earlier code actually did use final types and then I learned just how buggy older C++ 11 compilers can be.

Reply:
Good catch, I refactored the trait implementation to only work with the domain and storage types directly. Is that sufficient?

Reply:
Unfortunately not, `T::domain_type` still needs `T` to be fully known to the compiler.

Reply:
This is in latest isn't it?

`template <class From, class ToValueType> using is_type_erasable_to = detail::is_type_erasable_to<typename detail::remove_cvref<From>::type::domain_type, ToValueType>;`

Reply:
Yes and no, that's the public API which I don't use in the constructor constraints.

Reply:
Don't worry, I didn't like the naming ambiguity from the beginning.